services/concept_insights/ci.js

```diff
@@ -0,0 +1,422 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+module.exports = function (RED) {
+  var request = require('request');
+  var cfenv = require('cfenv');
+  var watson = require('watson-developer-cloud');
+
+  var username, password;
+  var service = cfenv.getAppEnv().getServiceCreds(/concept insights/i);
+
+  if (service) {
+    username = service.username;
+    password = service.password;
+  }
+
+  RED.httpAdmin.get('/watson-concept-insights/vcap', function (req, res) {
+    res.json(service ? {bound_service: true} : null);
+  });
+
+  function CorpusConfigurationNode(config) {
+    RED.nodes.createNode(this,config);
+    this.cname = config.cname;
+    this.access = config.access;
+    this.publiccorpus = config.publiccorpus;
+  }
+
+  RED.nodes.registerType("watson-concept-insights-corpus", CorpusConfigurationNode);
+
+  function searchNode(config) {
+    RED.nodes.createNode(this, config);
+    var node = this;
+
+    this.on('input', function(msg) {
+      setupConceptInsightsNode(msg,config,this,function(concept_insights,account_id) {
+        this.corpus = RED.nodes.getNode(config.corpus);
+        
+        if (this.corpus) {
+          var corpus_name;
+          if (this.corpus.cname) {
+            corpus_name = this.corpus.cname;
+          } else {
+            console.log(this.corpus);
+            corpus_name = this.corpus.publiccorpus;
+            account_id = 'public';
+          }        
+          var document_name = config.docname;
+
+          node.status({fill:"blue",shape:"ring",text:"requesting..."});
+
+          //Check for document or corpus mode
+          if (config.mode === 'document') {
+
+            if (!document_name) {
+              node.status({});
+              var message = 'No document name specified';
+              return node.error(message, msg);
+            }
+
+            var params = {
+              id: '/corpora/'+account_id+'/'+corpus_name+'/documents/'+document_name
+            }
+
+            switch (config.docapicall) {
+              case 'annotations':
+                params.id = params.id+'/annotations';
+                concept_insights.corpora.getDocumentAnnotations(params, function(err,res) {
+                  handleWatsonCallback(node,msg,err,res);
+                });
+                break;
+
+              case 'relatedconcepts':
+                concept_insights.corpora.getRelatedConcepts(params, function(err,res) {
+                  handleWatsonCallback(node,msg,err,res);
+                });
+                break;
+
+              case 'relationscores':
+              params.concepts = msg.concepts;
+                concept_insights.corpora.getRelationScores(params, function(err,res) {
+                  handleWatsonCallback(node,msg,err,res);
+                });
+                break;
+            }
+          } else {
+            //Corpus mode
+            var params = {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Variable 'params' is already declared in line 73.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2392368955/issues/source?bid=3225115&fileBranchId=3228814#l100)